qiskit_machine_learning/connectors/torch_connector.py

```diff
@@ -172,7 +173,7 @@ def backward(ctx: Any, grad_output: Tensor) -> Tuple:  # type: ignore
                 # pass) j for each observation i in the batch. Multiplies this with the gradient
                 # from this point on backwards with respect to each input k. Sums over all i and
                 # j to get total gradient of output w.r.t. each input k.
-                input_grad = einsum("ij,ijk->k", grad_output, input_grad)
+                input_grad = einsum("ij,ijk->ik", grad_output, input_grad)
```

Comment:
I changed the comment to try to explain this. Basically the input gradients depend on the batch dimension, but the weight gradients depend only on the layer parameter shape you determine at the beginning, that is why I added the index ` i` only for the `input_grad`. Without this, the dimensions don't match during  backprop for batch sizes larger than 1.

Reply:
In case you were wondering, the unit tests give the same results because during the check you end up summing all the gradients anyway.

Reply:
I see. But when gradients were computed without `i` I did not see an error in backdrop.

Reply:
Yes, I was also wondering how did the MNIST tutorial manage to run originally. My theory is that it's related to the new PyTorch release (1.9.0), that came out in June 15. 

Before, I think that torch was broadcasting the input_grad tensors to "compensate" for the missing batch dimension during backprop. And because the original tutorial only used batch size 1, everything worked fine. The new PyTorch release has changed some aspects of broadcasting, so it's most likely not possible to do this anymore (and I think that it would not have worked well for batch size >1 anyway).  I don't know if there is any test with hybrid networks and batch size >=2, but I haven't found it. Actually, now that I think about it, it would make a lot of sense to include a few hybrid nn unit tests...